c10/core/TensorImpl.h

```diff
@@ -1092,9 +1092,7 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
   */
   void _set_zero(bool value) {
     if (value) {
-      TORCH_INTERNAL_ASSERT(
-          false,
-          "Please call `torch._efficientzerotensor` if you want to create a tensor with no storage.");
+      key_set_ = key_set_.add(DispatchKey::ZeroTensor);
```

Comment:
If you don't want to change this, you can also add this key to the DispatchKeySet you create in the factory functions. Why relax this here?

Reply:
sure